### PR TITLE
 Added Navigation Bar , Search Bar and Genres

### LINKED ISSUE #13: Design: improve UI of the app
### Description:
Enhance the app's navigation by adding a bottom navigation bar with two primary tabs: "Dashboard" and "Playlist Add/Remove View". Also create a Dashboard UI, which includes a textbox for entering a genre or mood and a button to initiate playlist selection.

### Tips for the issue:
- feel free to give a new look to the app

### To do:
- [x] Ask us to assign the issue.
- [x] Once the issue is assigned, you can start working on it.
- [x] Create a PR
 
### Resources:
- [Flutter Docs on Navigation](https://docs.flutter.dev/ui/navigation)
- [Flutter app UI Docs](https://docs.flutter.dev/ui)

### Note:
The task is **assigned to contributors on a first-come, first-serve basis**, and the contributor has to **report progress every 3 days** to ensure active development on the tasks. If you need any help feel free to tag and message

### PR DESCRIPTION
**Pull request by D-extremity**
closes #13
Design: improve UI of the app #13
Navigation Bar is now working with attractive effect
Search bar has been added to Search Screen 
8 Genre item List has been added
Layout of Library has been added
Screenshot_1697316373
Screenshot_1697316373
Screenshot_1697316363
Screenshot_1697316358

### CODE REVIEW

**Comment by D-extremity** [comment]
@dk-a-dev merge it please.

**Comment by D-extremity** [comment]
@shubhbansal9 merge it please🙌🏻✨

File: pubspec.yaml
Comment (dk-a-dev) [inline pubspec.yaml:1]: > - assets/

also works

Comment (D-extremity) [reply pubspec.yaml:1]: Sorry , i didn't get it , do i need to make some changes? @dk-a-dev

File: lib/main.dart [hunk 1]
```diff
@@ -10,12 +12,15 @@ class MyApp extends StatelessWidget {
...
   Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Playtify',
-      theme: ThemeData(
-        primarySwatch: Colors.blue,
+    return ChangeNotifierProvider<RecentlyPlayedLogic>(
+      create: (_) => RecentlyPlayedLogic(),
+      child: MaterialApp(
+        title: 'Playtify',
+        theme: ThemeData(
```
Comment (dk-a-dev) [inline lib/main.dart:19]: switch to material-3
Add ..
> useMaterial3: true

refer [Material-3-Flutter](https://docs.flutter.dev/ui/design/material)

File: lib/home_screen.dart [hunk 2]
```diff
@@ -14,7 +16,11 @@ class _HomeScreenState extends State<HomeScreen> {
   int selectedIndex = 0;
   static const TextStyle optionStyle =
```
Comment (dk-a-dev) [inline lib/home_screen.dart:17]: seems like an unused field please remove it

Comment (D-extremity) [reply lib/home_screen.dart]: all changes done... @dk-a-dev

File: lib/pages/playlist.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+class _PlayListPageState extends State<PlayListPage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Padding(
+        padding: const EdgeInsets.all(8),
+        child: Column(
+          children: [
+            Image.network(
```
Comment (dk-a-dev) [inline lib/pages/playlist.dart:18]: image

Problem of image overflow
there are many ways to fix it, learn how to fix this
also, see what the point of adding an image there I don't get it.

File: lib/home_screen.dart
Comment (dk-a-dev) [inline lib/home_screen.dart:1]: image
image
the alignment of elements at bottom nav bar seems wrong try something new see material 3 flutter page for ideas
also see pinterest and dribble for app design and try to implement something new

Comment (D-extremity) [reply lib/home_screen.dart:1]: actually  type: BottomNavigationBarType.shifting, this line of code gives it effect whatever is being tapped will get somewhat larger and other will shift little bit opposite direction of it .

**Comment by dk-a-dev** [comment]
image
image

One more thing, adding images for selection for genre and moods is a good idea but... see the issue first those are fixed and changing screen sizes impact on them
try using cards , add text of mood for identification, add glass effect on carded images background, or try something new loom for ideas on google, pintrest or dribble

**Comment by dk-a-dev** [comment]
I will soon be adding lint rules(will inform you) that will enforce good coding practices and your editor will help u for fixes
Tip of the Day: 
- cntl+'.' for quick widget tips
image
- shift+alt+f5 for auto alignment of code

**Comment by dk-a-dev** [comment]
@shubhbansal9 why u merge this one?? It was under review

**Pull request merged**